### PR TITLE
Fix ansible 2.5.5 w/ python 2.7 image build

### PR DESCRIPTION
**Pull request by Mallear**
Dockerfile re-arrangement for build passing. This is a quickfix for #172 to pass, this image will need to be upgrade to python 3.x and Ansible 2.7 at least.

### CODE REVIEW

**Comment by Crow-EH** [comment]
Hi @Mallear and thanks for your contribution !

I think you'll have to fix both ansible and java 6 in the same PR for the checks to pass.

**Comment by Crow-EH** [comment]
Also, can you specify in the changelog that you changed the base image for ansible please ?

**Comment by fpavageau** [comment]
@Crow-EH we can merge anyway, as long as we can observe that the Ansible build is fixed. Let's see what happens in Travis.

File: ansible/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,9 @@
...
 
 WORKDIR /root
 RUN echo "Install Ansible with deps" && \
-    apk --update --no-cache add bash ca-certificates curl git openssh-client python=${PYTHON_VERSION} py-pip ansible=${VERSION} && \
-    echo "Done install Ansible with deps" &&\
-    echo "Install CI Helper" && \
-    curl -sSL https://github.com/rande/gitlab-ci-helper/releases/download/${CI_HELPER_VERSION}/alpine-amd64-gitlab-ci-helper -o /usr/bin/ci-helper && \
-    chmod 755 /usr/bin/ci-helper && \
-    echo "Done install CI Helper"
+    apk add --update -q --no-cache bash build-base curl git libffi-dev make openssh-client rsync tzdata openssl-dev && \
```
Comment (fpavageau) [inline ansible/Dockerfile]: What are `build-base`, `libffi-dev`, `make` and `openssl-dev` for? Can they be removed after `pip install`?

Maybe also remove the Alpine index files at the end?

Comment (Mallear) [reply ansible/Dockerfile]: Nop, we need this to be able to compile some python library using Cpython. As this image maybe use with other python script or python lib, we need these libs directly in the source image.

**Comment by Mallear** [comment]
Fix build, I forgot to change Ansible version in travis.yml

File: ansible/Dockerfile (hunk 1 above)
Comment (fpavageau) [reply ansible/Dockerfile]: OK for the packages, but can we still add `rm -rf /var/cache/apk/*` at the end?

File: ansible/Dockerfile [hunk 2]
```diff
@@ -1,14 +1,9 @@
-FROM alpine:3.8
+FROM python:2.7-alpine3.8
 
-ARG CI_HELPER_VERSION
-ARG PYTHON_VERSION
-ARG VERSION
+ARG VERSION="2.5.5"
```
Comment (fpavageau) [inline ansible/Dockerfile]: Should the version have a default value?

File: .travis.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ services:
 
 matrix:
   include:
-    - env: LANGUAGE=ansible      VERSION=2.5.5-r0 PYTHON_VERSION=2.7.15-r1
+    - env: LANGUAGE=ansible      VERSION=2.5.5 PYTHON_VERSION=2.7.15-r1
```
Comment (fpavageau) [inline .travis.yml]: `PYTHON_VERSION` is useless now.

File: ansible/Dockerfile (hunk 2 above)
Comment (Mallear) [reply ansible/Dockerfile]: Deleted

File: .travis.yml (hunk 3 above)
Comment (Mallear) [reply .travis.yml]: Deleted

Comment (fpavageau) [reply .travis.yml]: Nope, it's still there in `.travis.yml`, you only removed its use in `travis.py`.

File: ansible/Dockerfile [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
 WORKDIR /root
 RUN echo "Install Ansible with deps" && \
-    apk --update --no-cache add bash ca-certificates curl git openssh-client python=${PYTHON_VERSION} py-pip ansible=${VERSION} && \
-    echo "Done install Ansible with deps" &&\
+    apk add --update -q --no-cache bash build-base curl git libffi-dev make openssh-client rsync tzdata openssl-dev && \
+    pip install ansible==${VERSION} && \
+    echo "Done install Ansible with deps" && \
     echo "Install CI Helper" && \
-    curl -sSL https://github.com/rande/gitlab-ci-helper/releases/download/${CI_HELPER_VERSION}/alpine-amd64-gitlab-ci-helper -o /usr/bin/ci-helper && \
+    curl -sSL https://github.com/rande/gitlab-ci-helper/releases/download/${CI_HELPER_VERSION}/linux-amd64-gitlab-ci-helper -o /usr/bin/ci-helper && \
```
Comment (fpavageau) [inline ansible/Dockerfile]: Why change the distribution? It's still an Alpine-based image.

Comment (Mallear) [reply ansible/Dockerfile]: Sorry, copy/paste failure :)

**Comment by Mallear** [comment]
It is OK for me @Crow-EH @fpavageau . Java 6 image is still failing in Ekino repo's pipeline, but is working on mine : https://travis-ci.org/Mallear/docker-buildbox/jobs/562491183 . Temporary problem with SSL certificate of maven mirror ?

**Comment by fpavageau** [comment]
@Mallear the build on your fork didn't do anything as there was no modification in that folder, and maybe because it also half-failed?
> fatal: ambiguous argument '4fa7e3e7c9eb..f90c79227f7d': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
Namespace(branch='fix/ansible', ci_helper_version='0.0.6', commit_range='4fa7e3e7c9eb...f90c79227f7d', event_type='push', language='java', modd='0.5', pull_request='false', tag='', travis='true', version='6') BuildInfo(language='java', version='6', ci_helper_version='0.0.6', commit_range='4fa7e3e7c9eb...f90c79227f7d', branch='fix/ansible', tag='', **pull_request=False,** event_type='push', is_travis=True, modd='0.5')

vs
> Namespace(branch='master', ci_helper_version='0.0.6', commit_range='fd58b80057180178485a5d8c916dd03f9c1296e8...f90c79227f7dcd30fb8fbf68b219fa590a96e263', event_type='pull_request', language='java', modd='0.5', pull_request='176', tag='', travis='true', version='6') BuildInfo(language='java', version='6', ci_helper_version='0.0.6', commit_range='fd58b80057180178485a5d8c916dd03f9c1296e8...f90c79227f7dcd30fb8fbf68b219fa590a96e263', branch='master', tag='', **pull_request=True,** event_type='pull_request', is_travis=True, modd='0.5')

Anyway, the Ansible build passed, so we can already merge this.

**Comment by Mallear** [comment]
@fpavageau  Oh yes, didn't read everything, sorry

@Crow-EH It's done

**Comment by Crow-EH** [comment]
@Mallear Yes, it seems to be in fact a "temporary" SSL fail (CA certificate missing).

Not really temporary since I don't think we'll be able to update Java 6's Debian minor version (I don't think it's maintained anymore), bringing with it the missing up-to-date CA certificate.

If we can and it works, cool. But if not there are two solutions :
* turning off security with -k (bad)
* adding the missing certificate (long) then  refreshing the certificates with `update-ca-certificates --fresh`

I can check what I can do.

WDYT @fpavageau ?

PS: By the way this MR itself is OK now so we can merge it.

**Pull request merged**